src/plots/cartesian/axis_autotype.js

```diff
@@ -81,3 +82,10 @@ function category(a) {
 
     return curvecats > curvenums * 2;
 }
+
+function multiCategory(a) {
+    return (
+        Array.isArray(a[0]) && Array.isArray(a[1]) &&
+        (category(a[0]) || category(a[1]))
+    );
```

Comment:
This is we detect multicategory trace used to auto-type the axes. I think someone thought of a better way during yesterday's meeting. I forgot what that was though.

Reply:
:zap: that second line.

Reply:
**Edit:** Removing that `&& (category(a[0]) || category(a[1])` breaks some `carpet` plots which allow 2D `x` and `y` arrays.

Reply:
.. so should we make a special case for `carpet` traces? Or require that one of the two inner arrays have categories.

Reply:
Carpet is definitely special, nobody would ever really mistake carpets for real multicategory. But for extensibility perhaps a trace module category “noMulticategory” or something.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3300/commits/3746e03dc9e970062a307c0f348d57573b056d63